backend/src/blueprints/payment/stripe_service.py

```diff
@@ -0,0 +1,63 @@
+from flask import Blueprint, request
+import os
+import stripe
+
+bp_url_prefix = "/payment"
+stripe_service = Blueprint("stripe_service", __name__, url_prefix=bp_url_prefix)
+
+stripe.api_key = os.getenv("STRIPE_API_KEY")
+endpoint_secret = os.getenv("STRIPE_ENDPOINT_SECRET")
+client_origin_url = os.getenv("CLIENT_ORIGIN_URL")
+
+
+@stripe_service.route("/webhook", methods=["POST"])
+def stripe_webhook():
+    payload = request.get_data(as_text=True)
+    sig_header = request.headers.get("Stripe-Signature", type=str)
+
+    try:
+        event = stripe.Webhook.construct_event(payload, sig_header, endpoint_secret)
+
+    except ValueError as e:
+        # Invalid payload
+        return "Invalid payload", 400
+    except stripe.error.SignatureVerificationError as e:
+        # Invalid signature
+        return "Invalid signature", 400
+
+    # Handle the checkout.session.completed event
+    if event["type"] == "checkout.session.completed":
+        print("Payment was successful.")
+        # TODO: run some custom code here
+        user_id = event["data"]["object"]["client_reference_id"]
+        print("post payment id: ", user_id)
+
+    return "Success", 200
+
+
+@stripe_service.route("/create-checkout-session", methods=["POST"])
+def create_checkout_session():
+    authId = request.headers.get("Auth-Token", type=str)
```

Comment:
Auth-Token was just what I named the header on the frontend. The name could be anything but the value there is "sub" from auth0. I can change the name if it makes the code more readable.

Reply:
Oh, I didn't see that you were including it as a custom header. We could keep it that way, or it might just be simpler to use `g.user_id` and omit the extra header entirely, if we plan on using the auth guard here anyways.